cmd/bbolt/main.go

```diff
@@ -1394,20 +1394,43 @@ func (cmd *benchCommand) runReadsSequentialNested(db *bolt.DB, options *BenchOpt
 	})
 }
 
-func checkProgress(results *BenchResults, finishChan chan interface{}, stderr io.Writer) {
-	ticker := time.Tick(time.Second)
+// StartProgress starts a goroutine to report progress periodically.
+func (r *BenchResults) StartProgress(w io.Writer) *BenchResults {
+	r.stop = make(chan struct{})
+	r.report = make(chan struct{})
+	go r.checkProgress(w)
+	return r
+}
+
+// StopProgress stops the goroutine that reports progress periodically.
+func (r *BenchResults) StopProgress() {
+	close(r.stop)
+}
+
+// ReportProgress immediately without waiting for the next tick.
+func (r *BenchResults) ReportProgress() {
+	select {
+	case r.report <- struct{}{}:
+	default:
+	}
+}
```

Comment:
```suggestion
```

Reply:
If you feel like these progress related methods should not belong to `BenchResults` struct, I can move them under a new type. Please let me know.